x-pack/plugins/infra/server/saved_objects/custom_dashboards/custom_dashboards_saved_object.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsFieldMapping, SavedObjectsType } from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import { schema, Type } from '@kbn/config-schema';
+import { InfraCustomDashboard } from '../../../common/custom_dashboards';
+
+export const INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE = 'infra-custom-dashboards';
+
+const properties: Record<keyof InfraCustomDashboard, SavedObjectsFieldMapping> = {
+  dashboardSavedObjectIdList: { type: 'keyword' },
+  assetType: { type: 'keyword' },
+  kuery: { type: 'text' },
+};
+const createSchema: Record<keyof InfraCustomDashboard, Type<any>> = {
+  dashboardSavedObjectIdList: schema.arrayOf(schema.string()),
+  assetType: schema.string(),
+  kuery: schema.maybe(schema.string()),
+};
+
+export const infraCustomDashboardsSavedObjectType: SavedObjectsType = {
+  name: INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE,
+  hidden: false,
+  namespaceType: 'multiple',
+  mappings: {
+    properties,
+  },
+  management: {
+    importableAndExportable: true,
+    icon: 'infraApp',
+    getTitle: () =>
+      i18n.translate('xpack.infra.infraAssetCustomDashboards.', {
+        defaultMessage: 'Infrastructure Asset Custom Dashboards',
+      }),
+  },
+  modelVersions: {
+    '1': {
+      changes: [],
+      schemas: {
+        create: schema.object(createSchema),
```

Comment:
nit: Could be just me but I find it easier to read the code when the schema is defined here
ex
```suggestion
       create: schema.object({
          dashboardSavedObjectIdList: 
          schema.arrayOf(schema.string()),
          assetType: schema.string(),
           kuery: schema.maybe(schema.string()),
      }),
```

Reply:
True, that would be more readable. The reason I've put it into a separate object is to have an explicit type `Record<keyof InfraCustomDashboard, Type<any>>` to make sure schema fields stay the same as the corresponding `InfraCustomDashboard` type. Tbh, I'd trade off some readability for more type-safety, or maybe there is a way to achieve both that I'm missing?